### PR TITLE
chore: feedback for @maurafortino 's #587 PR

### PR DESCRIPTION
**Pull request by denopink**

### CODE REVIEW

**Comment by denopink** [comment]
Went over the code with marua on a slack call. We'll merge this either today or tomorrow morning.

File: internal/sink/sink.go [hunk 1]
```diff
@@ -77,13 +74,19 @@ type Kafka struct {
...
-func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
+func NewSink(c Config, logger *zap.Logger, listener ancla.Register) (Sink, error) {
 	switch l := listener.(type) {
 	case *ancla.RegistryV1:
 		v1 := &WebhookV1{}
 		v1.Update(c, logger, l.Registration.Config.AlternativeURLs, l.GetId(), l.Registration.FailureURL, l.Registration.Config.ReceiverURL)
-		return v1
+		return v1, nil
 	case *ancla.RegistryV2:
+		if len(l.Registration.Kafkas) == 0 && len(l.Registration.Webhooks) == 0 {
```
Comment (maurafortino) [inline internal/sink/sink.go:84]: would it make sense to add this to the validation portion of web hook-schema? just thinking about ways to try and clean up the this switch statement

File: internal/sink/sink.go [hunk 2]
```diff
@@ -97,33 +100,37 @@ func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
 						deliveryInterval: c.DeliveryInterval,
 						deliveryRetries:  c.DeliveryRetries,
 					},
+					client: &http.Client{},
 				}
+				if len(wh.ReceiverURLs) == 0 && len(wh.DNSSrvRecord.FQDNs) == 0 {
```
Comment (maurafortino) [inline internal/sink/sink.go:105]: same with this?

File: internal/sink/sink.go [hunk 3]
```diff
@@ -97,33 +100,37 @@ func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
...
-						err := dialer.lookup(domain, sortBy)
-						if err != nil {
-							v2.logger.Error("error received looking up service records", zap.Error(err))
-						}
-						dialer.NewClient()
-						v2.dialers = append(v2.dialers, dialer)
-					}
 				}
+
+				transport, err := NewSRVRecordDailer(wh.DNSSrvRecord)
```
Comment (maurafortino) [inline internal/sink/sink.go:119]: I know it was already mentioned but just making a note so we remember to change this to NewSRVRecordDialer

File: internal/sink/sink.go [hunk 4]
```diff
@@ -752,11 +760,8 @@ func (v2 *WebhookV2) send(secret, acceptType string, msg *wrp.Message) error {
...
 
 	var resp *http.Response
-	if len(v2.dialers) != 0 {
-		for _, d := range v2.dialers {
-			resp, err = d.client.Do(req)
-		}
-
+	if v2.client != nil {
+		resp, err = v2.client.Do(req)
 	} else {
```
Comment (maurafortino) [inline internal/sink/sink.go:765]: do we want this else statement? are we not trying to have retry logic for the custom client that we have?

File: internal/sink/srvRecordDailer.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	case "weight":
+		sort.Slice(d.srvs, func(i, j int) bool {
+			return d.srvs[i].Weight > d.srvs[j].Weight
+		})
+	case "priortiy":
+		sort.Slice(d.srvs, func(i, j int) bool {
+			return d.srvs[i].Priority < d.srvs[j].Priority
+		})
+	default:
+		return nil, fmt.Errorf("unknwon loadBalancingScheme type: %s", d.dnsSrvRecord.LoadBalancingScheme)
```
Comment (maurafortino) [inline internal/sink/srvRecordDailer.go:50]: spelling change: `unknown`

File: internal/sink/sink.go [hunk 6]
```diff
@@ -132,21 +139,22 @@ func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
...
-			return sink
+
+			return sink, nil
 		}
-	default:
-		return nil
 	}
-	return nil
+
+	return nil, fmt.Errorf("unknow webhook registry type")
```
Comment (maurafortino) [inline internal/sink/sink.go:157]: spelling error: `unknown`

File: internal/sink/sink.go (hunk 4 above)
Comment (maurafortino) [reply internal/sink/sink.go:765]: i should be able to work on this

**Pull request merged**